scripts/ecs/appsignals/setup-ecs-demo.sh

```diff
@@ -0,0 +1,1124 @@
+  adot_java_image="public.ecr.aws/aws-observability/adot-autoinstrumentation-java:$adot_java_image_tag"
+  sed -i '' "s|\"adot-java-image\"|\"${adot_java_image}\"|" ./sample-app/task-definitions/spring-petclinic-admin-server.json
+  sed -i '' "s|\"admin-server-image\"|\"${ECR_IMAGE_PREFIX}/springcommunity/spring-petclinic-admin-server\"|" ./sample-app/task-definitions/spring-petclinic-admin-server.json
+  sed -i '' "s|us-west-2|${REGION}|g" ./sample-app/task-definitions/spring-petclinic-admin-server.json
+  sed -i '' "s|000111222333|${ACCOUNT_ID}|g" ./sample-app/task-definitions/spring-petclinic-admin-server.json
+
+  aws ecs register-task-definition --cli-input-json file://sample-app/task-definitions/spring-petclinic-admin-server.json > /dev/null
+
+  namespace_id=$(aws servicediscovery list-namespaces --query "Namespaces[?Name=='ecs-petclinic'].Id" --output text)
+  service_discovery_id=$(aws servicediscovery create-service \
+      --name admin-server \
+      --dns-config "NamespaceId=$namespace_id,RoutingPolicy=WEIGHTED,DnsRecords=[{Type=A,TTL=300}]" \
+      --health-check-custom-config FailureThreshold=2 \
+      --query "Service.Id" --output text)
+
+  registryArn=$(aws servicediscovery get-service \
+    --id $service_discovery_id \
+    --query 'Service.Arn' --output text)
+
+  aws ecs create-service \
+      --service-name admin-server \
+      --task-definition admin-server \
+      --service-registries registryArn=$registryArn \
+      --desired-count 1 \
+      --launch-type FARGATE \
+      --network-configuration "awsvpcConfiguration={subnets=[$(echo $SUBNET_IDS | tr -s ' ' ',')],securityGroups=[$SECURITY_GROUP_ID],assignPublicIp=ENABLED}" \
+      --cluster $CLUSTER > /dev/null
+
+  sleep 120
+
+  echo "Finished deploying admin server!"
+
+}
+
+function run_api_gateway() {
+  api_gateway_target_group_arn=$(aws elbv2 create-target-group \
+      --name api-gateway-target-group \
+      --protocol HTTP \
+      --port 8080 \
+      --vpc-id $VPC_ID \
+      --target-type ip \
+      --query "TargetGroups[0].TargetGroupArn" \
+      --output text)
+
+  aws elbv2 modify-target-group \
+      --target-group-arn $api_gateway_target_group_arn \
+      --health-check-protocol HTTP \
+      --health-check-path "/" \
+      --health-check-interval-seconds 240 \
+      --health-check-timeout-seconds 60 \
+      --healthy-threshold-count 5 \
+      --unhealthy-threshold-count 2 > /dev/null
+
+  aws elbv2 create-listener \
+    --load-balancer-arn $LOAD_BALANCER_ARN \
+    --protocol HTTP \
+    --port 8080 \
+    --default-actions Type=forward,TargetGroupArn=$api_gateway_target_group_arn > /dev/null
+
+  adot_java_image_tag=$(curl -s -I -L 'https://github.com/aws-observability/aws-otel-java-instrumentation/releases/latest' | grep -i Location | awk -F'/tag/' '{print $2}' | tr -d '\r')
```

Comment:
Do we need to parse the adot image tag every time in the service creation? Can we get a finalized image at the beginning?

In very extreme case, we may end up having different ADOT java image for some applications during the release.

Reply:
Yes, updated to define the adot image at the beginning of the script.